core/src/main/java/org/mule/api/processor/util/InvokerMessageProcessorUtil.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.api.processor.util;
+
+import static org.apache.commons.lang.StringUtils.countMatches;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public final class InvokerMessageProcessorUtil
+{
+
+    private InvokerMessageProcessorUtil()
+    {
+
+    }
+
+    /**
+     * Split and merge comma separated mule expressions 
+     * when they falls within an array
+     * 
+     * @param arguments arguments string expression
+     * @return arguments merged
+     */
+    public static List<String> splitAndMergeArgumentsExpression(String arguments)
+    {
+        List<String> argumentsVerified = new ArrayList<String>();
+        String[] argumentsSplitted = arguments.split("\\s*,\\s*");
+        if (argumentsSplitted.length <= 1)
+        {
+            return Arrays.asList(argumentsSplitted);
+        }
+
+        return merge(argumentsVerified, argumentsSplitted);
+    }
+
+    private static List<String> merge(List<String> argumentsVerified, String[] argumentsSplitted)
+    {
+        StringBuffer mergingArg = new StringBuffer();
+        mergingArg.append(argumentsSplitted[0]);
+
+        for (int index = 1; index < argumentsSplitted.length; index++)
+        {
+            String arg = argumentsSplitted[index];
+            if (hasToBeMerged(mergingArg))
+            {
+                mergingArg.append(", " + arg);
```

Comment:
you're adding an extra space after the comma which may not have been there in the original expression